src/components/GoalPopupFormTrigger.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useRef, FC, ReactNode, MutableRefObject, useCallback } from 'react';
+import styled from 'styled-components';
+import { Popup, nullable } from '@taskany/bricks';
+import { gapS } from '@taskany/colors';
+
+const StyledPopup = styled(Popup)`
+    padding: ${gapS};
+`;
+
+interface GoalPopupFormTriggerProps {
+    children: ReactNode;
+    renderTrigger: (props: { onClick: () => void; ref: MutableRefObject<HTMLDivElement | null> }) => ReactNode;
+}
+
+export const GoalPopupFormTrigger: FC<GoalPopupFormTriggerProps> = ({ children, renderTrigger }) => {
+    const [visible, setVisible] = useState(false);
+    const popupRef = useRef<HTMLDivElement | null>(null);
+
+    const onClickOutside = useCallback(() => {
+        setVisible(false);
+    }, []);
+
+    return (
+        <>
+            {renderTrigger({
+                onClick: () => setVisible(true),
+                ref: popupRef,
+            })}
+            <StyledPopup
+                placement="bottom-start"
+                visible={visible}
+                reference={popupRef}
+                onClickOutside={onClickOutside}
+                interactive
+                minWidth={450}
+                maxWidth={450}
+                arrow
+            >
+                {nullable(visible, () => children)}
```

Comment:
Do this condition required here? U have same prop for Popup

Reply:
Popup does not unmount content by default, this condition allows you to do this.

We need this so that the initialized useState is updated when the popup is re-opened.